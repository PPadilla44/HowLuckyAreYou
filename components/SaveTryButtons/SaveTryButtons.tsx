import { StyleSheet } from 'react-native';
import React, { FC } from 'react';
import { Text, TouchableOpacity, View } from "../Themed";
import Colors from '../../constants/Colors';


interface Props {
    showTry: boolean
    handleTry: () => void;
    handleSave: () => void;
}


const SaveTryButtons: FC<Props> = ({ showTry, handleSave, handleTry }) => {

    return (
        <View style={styles.container}>

            <TouchableOpacity testID='saveBtn'
                disabled={!showTry}
                lightColor={showTry ? "" : Colors.dark.input}
                darkColor={showTry ? "" : Colors.dark.input}
                containerStyle={[styles.saveBtn, { opacity: showTry ? 1 : .5 } ]}
                onPress={handleSave}>
                <Text style={styles.saveText}>Save</Text>
            </TouchableOpacity>

            <TouchableOpacity
                testID='tryBtn'
                lightColor={showTry ? Colors.light.input : "transparent"}
                darkColor={showTry ? Colors.dark.input : "transparent"}
                containerStyle={styles.tryBtn}
                onPress={handleTry}
                disabled={!showTry}
            >
                <Text
                    lightColor={showTry ? "" : Colors.dark.input}
                    darkColor={showTry ? "" : Colors.dark.text}
                    style={[styles.tryText, { opacity: showTry ? 1 : .5 }]} >
                    Try
                </Text>
            </TouchableOpacity>

        </View>
    )
}

export default SaveTryButtons

const styles = StyleSheet.create({
    container: {
        flexDirection: "row",
        height: 60,
        backgroundColor: "transparent",
    },
    tryBtn: {
        flex: 2,
        alignItems: "center",
        justifyContent: "center",
    },
    tryText: {
        fontSize: 36,
        fontWeight: "bold"
    },
    saveBtn: {
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
    },
    saveText: {
        color: "white",
        fontSize: 36,
        fontWeight: "bold"
    }
})