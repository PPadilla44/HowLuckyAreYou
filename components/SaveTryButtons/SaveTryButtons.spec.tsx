import { render, RenderAPI, fireEvent } from "@testing-library/react-native";
import SaveTryButtons from "./SaveTryButtons";

describe('<SaveTryButtons />', () => {
    
    const saveMock = jest.fn();
    const tryMock = jest.fn();


    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(<SaveTryButtons showTry={false} handleSave={saveMock} handleTry={tryMock}  />)
    });

    it('should render correctly', () => {
        wrapper.getByText("Save");
        wrapper.getByText("Try");
        wrapper.getByTestId("saveBtn")
        wrapper.getByTestId("tryBtn")
    });

    it('should render correct disabled styles', () => {
        const tryBtn = wrapper.getByTestId("tryBtn");
        const tryText = wrapper.getByText("Try");
        wrapper.container.findByProps({ showTry: false });
        wrapper.container.findByProps({ lightColor: "transparent" });
        wrapper.container.findByProps({ darkColor: "transparent" });
        expect(tryBtn).toBeDisabled();
        expect(tryText).toHaveStyle({ opacity: .5 })
    });

    it('should render correct enabled styles', () => {
        wrapper.update(<SaveTryButtons showTry={true} handleSave={saveMock} handleTry={tryMock} />)
        const tryBtn = wrapper.getByTestId("tryBtn");
        const tryText = wrapper.getByText("Try");
        wrapper.container.findByProps({ showTry: true });
        wrapper.container.findByProps({ lightColor: "#E6E6E6" });
        wrapper.container.findByProps({ darkColor: "#808080" });
        expect(tryBtn).toBeEnabled();
        expect(tryText).toHaveStyle({ opacity: 1 })
    });

    it('should handle button presses', () => {
        fireEvent.press(wrapper.getByTestId("saveBtn"));
        expect(saveMock).toHaveBeenCalled()
        fireEvent.press(wrapper.getByTestId("tryBtn"));
        expect(tryMock).toHaveBeenCalled()
    });


    it('should match snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot()
    });

});