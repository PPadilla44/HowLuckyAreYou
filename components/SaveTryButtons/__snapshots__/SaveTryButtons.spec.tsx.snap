// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SaveTryButtons /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "shadowColor": "white",
      },
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
        "height": 60,
      },
    ]
  }
>
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "backgroundColor": "#325C99",
          "borderColor": "white",
        },
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
    testID="saveBtn"
  >
    <View
      accessible={true}
      style={
        Object {
          "backgroundColor": "#325C99",
          "borderColor": "white",
          "opacity": 1,
          "shadowColor": "white",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "Futura",
            },
            Object {
              "color": "white",
              "fontSize": 36,
              "fontWeight": "bold",
            },
          ]
        }
      >
        Save
      </Text>
    </View>
  </RNGestureHandlerButton>
  <RNGestureHandlerButton
    collapsable={false}
    exclusive={true}
    onGestureEvent={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onHandlerStateChange={[Function]}
    rippleColor={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderColor": "white",
        },
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        },
      ]
    }
    testID="tryBtn"
  >
    <View
      accessible={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderColor": "white",
          "opacity": 1,
          "shadowColor": "white",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "white",
              "fontFamily": "Futura",
            },
            Array [
              Object {
                "fontSize": 36,
                "fontWeight": "bold",
              },
              Object {
                "opacity": 0.5,
              },
            ],
          ]
        }
      >
        Try
      </Text>
    </View>
  </RNGestureHandlerButton>
</View>
`;
